### ISSUE TITLE
Gradient Accumulation causes lower lr to be same as non-gradient accumulation

### ISSUE THREAD

**Issue by marii-moe** [fastai/fastai#3023]
I am already working on this, will update this description as I fix it. We do not scale the loss by number of batches to accumulate, so we end up with too large of gradients when doing gradient accumulation. 

More details here: https://github.com/fastai/fastai2/pull/194

**Comment by marii-moe** [comment]
Noticed that `BnFreeze` also seems to not be working in the same notebook. Its only test is slow, so it probably wasn't caught when something else changed. Taking a look now.

**Comment by marii-moe** [comment]
@borisdayma just checking in as you were working on fastai/fastai2#194, want to review this one?

**Comment by borisdayma** [comment]
Sure let me know how I can help.
Gradients are added and not averaged but it makes sense to change it to make it easier to use. My recommendation here would be just to keep track of how many samples are added and divide at the end.
Also the documentation will need to be updated as it mentions that the learning rate needs to be decreased.

I was not aware of potential issues with batch norm so it's great you noticed it!
